### PR TITLE
Add Binomial distribution

### PR DESCRIPTION
**Pull request by fehiepsi**
Based on Multinomial distribution code, I implement Binomial distribution. I also change the code of Multinomial distribution a little bit to fix bugs when parameter `n` is an `int` (e.g. in `expanded_sample()` function, we can not get `self.n.data.dim()`.

### CODE REVIEW

**Comment by fehiepsi** [comment]
My mistake, I delete the wrong branch.

**Comment by fehiepsi** [comment]
It seems that travis failed because of exceeding time.

File: pyro/distributions/multinomial.py [hunk 1]
```diff
@@ -25,6 +25,11 @@ class Multinomial(Distribution):
     def __init__(self, ps, n, batch_size=None, *args, **kwargs):
         if ps.dim() not in (1, 2):
             raise ValueError("Parameter `ps` must be either 1 or 2 dimensional.")
+        if isinstance(n, int):
```
Comment (fritzo) [inline pyro/distributions/multinomial.py]: Prefer `isinstance(n, (int, long))` or `isinstance(n, numbers.Number)`.

Comment (fehiepsi) [reply pyro/distributions/multinomial.py]: Thanks! I prefer your second suggestion. In python 3, there is no `long` type.

**Pull request merged**